lib/fluent/supervisor.rb

```diff
@@ -354,13 +354,21 @@ def initialize(path, level, chuser, chgroup, opts, log_rotate_age: nil, log_rota
         @log_rotate_size = log_rotate_size
       end
 
+      def worker_id_suffixed_path(worker_id, path)
+        require 'pathname'
+
+        Pathname(path).sub_ext("-#{worker_id}#{Pathname(path).extname}").to_s
+      end
+
       def init(process_type, worker_id)
         @opts[:process_type] = process_type
         @opts[:worker_id] = worker_id
 
         if @path && @path != "-"
           @logdev = if @log_rotate_age || @log_rotate_size
-                     Fluent::LogDeviceIO.new(@path, shift_age: @log_rotate_age, shift_size: @log_rotate_size)
+                     Fluent::LogDeviceIO.new(Fluent.windows? ?
+                                               worker_id_suffixed_path(worker_id, @path) : @path,
```

Comment:
if the `process_type` is `:supervisor`, this method call looks unnecessary for me. supervisor's worker id is 0 so the name of log file for supervisor becomes `log-supervisor-0.log`, right?

Reply:
> if the `process_type` is `:supervisor`, this method call looks unnecessary for me.

Yeah, I think so too before testing for it.
If `log-supervisor-0.log` and `log-0.log are not separated, Fluentd still complains the following error:

```
log shifting failed. Permission denied @ rb_file_s_rename - ([Some sort of moving files])
log writing failed. closed stream
log shifting failed. closed stream
log writing failed. closed stream
log shifting failed. closed stream
log writing failed. closed stream
log shifting failed. closed stream
log writing failed. closed stream
log shifting failed. closed stream
log writing failed. closed stream
log shifting failed. closed stream
log writing failed. closed stream
```

I have no idea for this issue....

> supervisor's worker id is 0 so the name of log file for supervisor becomes log-supervisor-0.log, right?

Yes.

Reply:
> If log-supervisor-0.log and log-0.log are not separated,

I meant it was better the file name of a supervisor was `log-supervisor.log` not `log-0.log`.
If you have the intention to do it (adding `-supervisor` and `-0` to a supervisor file name), I'm ok with it since it's a pretty minor point :)

Reply:
> If you have the intention to do it (adding `-supervisor` and `-0` to a supervisor file name), I'm ok with it since it's a pretty minor point :)

Ahhh, OK. Their file names are intended.